### DISCUSSION TITLE
Couple of additional sensors for battery installations

### DISCUSSION THREAD

**Discussion by Daibutt** [macxq/foxess-ha#33] [Show and tell]
Posting these in case anyone else finds them useful.

- Battery life based on average daily load
- Battery life based on current load
- Battery stored energy in kWh

NB: If your min SoC is not the default 10% you'll have to change the '10's in the YAML below too.

```
- platform: template
  sensors:
    battery_hrs_average:
      friendly_name: "Battery life based on average load"
      value_template: >
          {% if (states('sensor.foxess_bat_soc') in ['unknown', 'unavailable'])%}
          {{ states('sensor.battery_hrs_average') }}
          {% else %}
```
   
 Replace the '20' below with your battery total capacity in kWh and '13.7' by your average daily load in kWh

          {{ ((states('sensor.foxess_bat_soc') | float - 10) / 100 * 20 / 13.7 * 24) | round(1) }}
          {% endif %}
      unit_of_measurement: "hours"
    battery_hrs_instant:
      friendly_name: "Battery life based on current load"
      value_template: >
          {% if (states('sensor.foxess_bat_soc') in ['unknown', 'unavailable'])%}
          {{ states('sensor.battery_hrs_instant') }}
          {% else %}
  
 Replace the '20' below with your battery total capacity in kWh
 
         {{ ((states('sensor.foxess_bat_soc') | float - 10) / 100 * 20 / states('sensor.foxess_load_power') | float) | round(1) }}
          {% endif %}
      unit_of_measurement: "hours"
    battery_energy:
      friendly_name: "Battery stored energy"
      value_template: >
          {% if (states('sensor.foxess_bat_soc') in ['unknown', 'unavailable'])%}
          {{ states('sensor.battery_energy') }}
          {% else %}

 Replace the '20' below with your battery total capacity in kWh

          {{ ((states('sensor.foxess_bat_soc') | float) / 100 * 20) | round(1) }}
          {% endif %}
      unit_of_measurement: "kWh"

**Comment by Daibutt** [comment]
Apologies for the formatting !

**Reply by williamjeccles** [reply]
good job thanks!
